src/main/java/com/aws/greengrass/util/Exec.java

```diff
@@ -314,6 +316,11 @@ public Exec withErr(Consumer<CharSequence> o) {
             throw new InterruptedException();
         }
         process = createProcess();
+
+        logger.debug("Created process with pid {}",
+                process instanceof ProcessImplForWin32 ? ((ProcessImplForWin32) process).getPid()
+                        : Processes.newPidProcess(process).getPid());
```

Comment:
instead of the casting around, let's just add a `getPid` method onto Exec and implement it in the subclasses